### PR TITLE
Add collectors and background service for measurements

### PR DESCRIPTION
**Pull request by adinauer**
## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

**Review by philipphofmann — COMMENTED** [review]
I just gave this a quick glance as it's a big PR. @adinauer, what's the current state of this PR?

File: sentry-android-core/src/main/java/io/sentry/android/core/internal/measurement/cpu/CpuBackgroundMeasurementCollector.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  @Override
+  @SuppressLint("NewApi")
+  public @Nullable Object collect() {
+    try {
+      // TODO 1-5 ms
+      String stat = readProcSelfStat();
+      Long clockTicks = parseClockTicks(stat);
+
+      // TODO 2-4 ms on first read, then cached 4 - 20 μs
+      List<Integer> maxFrequencies = CpuInfoUtils.getInstance().readMaxFrequencies();
```
Comment (adinauer) [inline sentry-android-core/src/main/java/io/sentry/android/core/internal/measurement/cpu/CpuBackgroundMeasurementCollector.java:37]: Doesn't seem to return useful data on my emulator, need to wait for new device to arrive to retest.

File: sentry-android-core/src/main/java/io/sentry/android/core/internal/measurement/cpu/CpuMeasurementCollector.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  @Override
+  protected @Nullable Map<String, MeasurementValue> onTransactionFinishedInternal(
+      @NotNull ITransaction transaction, @NotNull MeasurementContext context) {
+    HashMap<String, MeasurementValue> results = new HashMap<>();
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1
+        && startRealtimeNanos != null
+        && startElapsedTimeMs != null) {
+      // TODO SystemClock.elapsedRealtimeNanos() + Process.getElapsedCpuTime() take 9 - 15 μs
+      Long diffRealtimeNanos = SystemClock.elapsedRealtimeNanos() - startRealtimeNanos;
```
Comment (adinauer) [inline sentry-android-core/src/main/java/io/sentry/android/core/internal/measurement/cpu/CpuMeasurementCollector.java:63]: This seems like our best bet as it's very fast and can be done synchronously. Probably also less risk of being taken away than reading `/proc/self/stat`

NOTE: The difference for `Process.getElapsedCpuTime()` might be higher than for `SystemClock.elapsedRealtimeNanos()` if multiple cores are under load.

**Comment by romtsn** [comment]
superseded by #2445

**Pull request closed without merging**